src/solver/conv_asm_implicit_gemm_gtc_bwd.cpp

```diff
@@ -0,0 +1,323 @@
+    int dilation_h = ctx.kernel_size_h > 1 ? ctx.kernel_dilation_h : 1;
+    int dilation_w = ctx.kernel_size_w > 1 ? ctx.kernel_dilation_w : 1;
+    int pad_h      = ctx.pad_h;
+    int pad_w      = ctx.pad_w;
+    int y          = ctx.kernel_size_h;
+    int x          = ctx.kernel_size_w;
+
+    int gcd_stride_dilation_h = gcd(stride_h, dilation_h);
+    int gcd_stride_dilation_w = gcd(stride_w, dilation_w);
+    int y_tilda               = stride_h / gcd_stride_dilation_h;
+    int x_tilda               = stride_w / gcd_stride_dilation_w;
+
+    int h_tilda = ho + (dilation_h * (y - 1) + stride_h - 1) / stride_h;
+    int w_tilda = wo + (dilation_w * (x - 1) + stride_w - 1) / stride_w;
+
+    int y_dot = integer_divide_ceil(y, y_tilda);
+    int x_dot = integer_divide_ceil(x, x_tilda);
+
+    int h_tilda_left = std::max(0, pad_h - dilation_h * (y_tilda - 1)) / stride_h;
+    int w_tilda_left = std::max(0, pad_w - dilation_w * (x_tilda - 1)) / stride_w;
+
+    int h_tilda_right = std::min(h_tilda, (pad_h + hi - 1 + stride_h - 1) / stride_h + 1);
+    int w_tilda_right = std::min(w_tilda, (pad_w + wi - 1 + stride_w - 1) / stride_w + 1);
+
+    int h_tilda_slice = h_tilda_right - h_tilda_left;
+    int w_tilda_slice = w_tilda_right - w_tilda_left;
+    int num_of_gemm   = y_tilda * x_tilda;
+    // clang-format on
+    int gemm_m = c;
+    int gemm_n = n * h_tilda_slice * w_tilda_slice;
+
+    for(; pConfig != tunables.end(); pConfig++)
+    {
+        if(pConfig->nxe == 0)
+        {
+            if((x != 1) || (y != 1) || (stride_h != 1) || (stride_w != 1) || (dilation_h != 1) ||
+               (dilation_w != 1) || (pad_h != 0) || (pad_w != 0))
+            {
+                continue;
+            }
+        }
+        if((gemm_n % pConfig->gemm_n_per_block != 0) || (gemm_m % pConfig->gemm_m_per_block != 0) ||
+           (k % pConfig->gemm_k_per_block != 0))
+        {
+            continue;
+        }
+        if(pConfig->gemm_n_per_block % pConfig->nxb != 0)
+        {
+            continue;
+        }
+        //# ho * wo is 4x, gemm_n is 256, hence need batch size 256/4=64x
+        if(n % (pConfig->gemm_n_per_block / pConfig->nxb) != 0)
+        {
+            continue;
+        }
+        if((h_tilda_slice * w_tilda_slice) % pConfig->nxb != 0)
+        {
+            continue;
+        }
+        bool gemm_k_valid = true;
+        for(int gemm_id = 0; gemm_id < num_of_gemm; gemm_id++)
+        {
+            int i_y_tilda          = gemm_id / x_tilda;
+            int i_x_tilda          = gemm_id % x_tilda;
+            int y_dot_slice        = (i_y_tilda + 1) * y_dot <= y ? y_dot : y % y_dot;
+            int x_dot_slice        = (i_x_tilda + 1) * x_dot <= x ? x_dot : x % x_dot;
+            int gemm_k             = k * y_dot_slice * x_dot_slice;
+            bool is_gemm_not_empty = gemm_k > 0;
+            if(is_gemm_not_empty)
+            {
+                if(gemm_k % pConfig->gemm_k_per_block != 0)
```

Comment:
If you need to check if gemm_k is divided gemm_k_per_block without remainder here, you could delete the check above.

Reply:
Should not call MIOPEN_THROW when no tunable is found

Reply:
@shaojiewang @fronteer , you are all right, I'll fix.